component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java

```diff
@@ -332,26 +339,74 @@ public int getSize() throws Exception {
     }
   }
 
+  private Query getIdentitiesQueryCount(String providerId) {
+    // Oracle and MSSQL support only 1/0 for boolean, Postgresql supports only
+    // TRUE/FALSE, MySQL supports both
+    String dbBoolFalse = isOrcaleDialect() || isMSSQLDialect() ? "0" : "FALSE";
+    String dbBoolTrue = isOrcaleDialect() || isMSSQLDialect() ? "1" : "TRUE";
+    // Oracle Dialect in Hibernate 4 is not registering NVARCHAR correctly, see
+    // HHH-10495
+    StringBuilder queryStringBuilder = null;
+    if (isOrcaleDialect()) {
+      queryStringBuilder = new StringBuilder("SELECT COUNT(DISTINCT to_char(identity_1.remote_id))\n");
+    } else if (isMSSQLDialect()) {
+      queryStringBuilder = new StringBuilder("SELECT COUNT(DISTINCT try_convert(varchar(200), identity_1.remote_id)) as remote_id  \n");
+    } else {
+      queryStringBuilder = new StringBuilder("SELECT COUNT(DISTINCT identity_1.remote_id)\n");
```

Comment:
queryStringBuilder = new StringBuilder("SELECT COUNT(DISTINCT identity_1.identity_id)\n")
it can be used regardless the DB

Reply:
Done!